source/CAS/Client.php

```diff
@@ -2626,6 +2622,22 @@ private function _callback()
         throw new CAS_GracefullTerminationException();
     }
 
+    /**
+     * Check if application/xml or text/xml is pressent in HTTP_ACCEPT header values
+     * when return value is complex and contains attached q parameters.
+     * Example:  HTTP_ACCEPT = text/html,application/xhtml+xml,application/xml;q=0.9
+     * @return bool
+     */
+    private function isXmlResponse()
+    {
+        if (str_replace('application/xml', '', $_SERVER['HTTP_ACCEPT']) != 'application/xml') {
```

Comment:
Why is `strpos` not used here if you want to check whether your Accept header contains a string? I am a little bit confused by the code indeed (e.g. `str_replace('application/xml', '', 'application/html') == 'application/html' !== 'application/xml'`)